Source/WebCore/platform/text/cocoa/LocaleCocoa.mm

```diff
@@ -89,11 +89,12 @@
     double msec = dateComponents.millisecondsSinceEpoch();
     DateComponentsType type = dateComponents.type();
 
-    // "week" type not supported.
     ASSERT(type != DateComponentsType::Invalid);
+#if !HAVE(UICALENDARSELECTIONWEEKOFYEAR)
```

Comment:
I don't really like the idea of referencing a UIKit `HAVE` down in WebCore code that is UI-independent.

Could you also add an `ENABLE_INPUT_TYPE_WEEK_PICKER` flag for use in WebCore and in WebKit outside of the UI code? Eventually other platforms could also flip that on.